packages/cli/lib/ast-helper.js

```diff
@@ -0,0 +1,149 @@
+  IfStatement: 'IfStatement',
+  ForStatement: 'ForStatement',
+  WhileStatement: 'WhileStatement',
+  DoStatement: 'DoStatement',
+  VariableStatement: 'VariableStatement',
+  FunctionDeclaration: 'FunctionDeclaration',
+  ArrowFunction: 'ArrowFunction',
+  ImportDeclaration: 'ImportDeclaration',
+  StringLiteral: 'StringLiteral',
+  FalseKeyword: 'FalseKeyword',
+  NullKeyword: 'NullKeyword',
+  AnyKeyword: 'AnyKeyword',
+  NumericLiteral: 'NumericLiteral',
+  NoSubstitutionTemplateLiteral: 'NoSubstitutionTemplateLiteral',
+  TemplateExpression: 'TemplateExpression',
+};
+
+const RootNodesFindID = [
+  // Defaul format generated by lb4 model
+  [
+    tsArtifact.ClassDeclaration,
+    tsArtifact.PropertyDeclaration,
+    tsArtifact.Identifier,
+  ],
+  // Model JSON definition inside the @model decorator
+  [
+    tsArtifact.ClassDeclaration,
+    tsArtifact.Decorator,
+    tsArtifact.CallExpression,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.Identifier,
+  ],
+  // Model JSON static definition inside the Class
+  [
+    tsArtifact.ClassDeclaration,
+    tsArtifact.PropertyDeclaration,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.Identifier,
+  ],
+];
+const ChildNodesFindID = [
+  // Defaul format generated by lb4 model
+  [
+    tsArtifact.ClassDeclaration,
+    tsArtifact.PropertyDeclaration,
+    tsArtifact.Decorator,
+    tsArtifact.CallExpression,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.TrueKeywordTrue,
+  ],
+
+  // Model JSON definition inside the @model decorator
+  [
+    tsArtifact.ClassDeclaration,
+    tsArtifact.Decorator,
+    tsArtifact.CallExpression,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.TrueKeywordTrue,
+  ],
+  // Model JSON static definition inside the Class
+  [
+    tsArtifact.ClassDeclaration,
+    tsArtifact.PropertyDeclaration,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.ObjectLiteralExpression,
+    tsArtifact.PropertyAssignment,
+    tsArtifact.TrueKeywordTrue,
+  ],
+];
+
+/**
+ * Parse the file using the possible formats specified in the arrays
+ * rootNodesFindID and childNodesFindID
+ * @param {string} fileContent with a model.ts class
+ */
+exports.getIdFromModel = async function(fileContent) {
+  let nodePos = 0;
+  let retVal = null;
+
+  for (let rootNodes of RootNodesFindID) {
+    const ast = tsquery.ast(fileContent);
```

Comment:
Is it necessary to build a new `ast` object for every `rootNodes` entry? Cannot we share the same `ast` objectt for all queries?

I am concerned about performance, parsing source code into AST representation takes non-negligible time from my experience with running `prettier` on loopback-next.